Lib/ntpath.py

```diff
@@ -41,14 +42,41 @@ def _get_bothseps(path):
 # Other normalizations (such as optimizing '../' away) are not done
 # (this is done by normpath).
 
-def normcase(s):
-    """Normalize case of pathname.
+try:
+    from _winapi import (
+        LCMapStringEx as _LCMapStringEx,
+        LOCALE_NAME_INVARIANT as _LOCALE_NAME_INVARIANT,
+        LCMAP_LOWERCASE as _LCMAP_LOWERCASE)
+
+    def normcase(s):
+        """Normalize case of pathname.
+
+        Makes all characters lowercase and all slashes into backslashes.
+        """
+        if isinstance(s, bytes):
+            if s == b'':
+                return s
+            s = os.fsdecode(s).replace('/', '\\')
+            return os.fsencode(_LCMapStringEx(_LOCALE_NAME_INVARIANT,
+                                              _LCMAP_LOWERCASE, s))
+        else:
+            if  isinstance(s, str) and s == '':
+                return s
+            s = os.fspath(s)
+            return _LCMapStringEx(_LOCALE_NAME_INVARIANT,
+                                  _LCMAP_LOWERCASE,
+                                  s.replace('/', '\\'))
```

Comment:
Sorry, I forgot the `os.fspath()` call in my last suggestion. The result of `os.fspath()` is either `str` or `bytes`.
```suggestion
   def normcase(s):
        """Normalize case of pathname.
        Makes all characters lowercase and all slashes into backslashes.
        """
        s = os.fspath(s)
        if not s:
            return s
        if isinstance(s, bytes):
            s = os.fsdecode(s).replace('/', '\\')
            return os.fsencode(_LCMapStringEx(_LOCALE_NAME_INVARIANT,
                                              _LCMAP_LOWERCASE, s))
        else:
            return _LCMapStringEx(_LOCALE_NAME_INVARIANT,
                                  _LCMAP_LOWERCASE,
                                  s.replace('/', '\\'))
```